torch/_inductor/compile_fx.py

```diff
@@ -541,7 +541,7 @@ def fx_codegen_and_compile(
                     if hasattr(out, "layout"):
                         output_strides.append(
                             tuple(
-                                V.graph.sizevars.size_hint(s) for s in out.layout.stride
+                                V.graph.sizevars.size_hint(s, fallback=2) for s in out.layout.stride
```

Comment:
This is because unbacked symbol appears here.

For example, running:
```shell
python test_torchinductor_dynamic_shapes.py -k test_item_unsqueeze_nobreak
```
out.layout is `FixedLayout('cpu', torch.float32, size=[i3, i3, 1], stride=[i3, 1, 1])`. I can only get a range from shape_env.var_to_range, so I give a default value.

Reply:
cc @shunting314 but this is pretty suspicious to me. I think to do this properly we need to infer the ordering instead of passing around hints. In the meantime, I think probably we should just figure out how to turn off layout optimization when something is unbacked. Maybe it suffices to just force the output to be None when its unbacked.

Reply:
yea, this part is a bit annoying. On a high level, I agree a stride order is better than stride hints. But there are some corner cases where a tensor's stride may not be 'dense'. mixnet_l has a tensor with size [8, 64, 112, 112] and strides (2408448, 1, 21504, 192) . Using stride order will miss some information and we can not recover the eager stride with that.

Reply:
This question may seem a bit naive, when will output_strides be used? I've noticed that force it to None when its unbacked doesn't seem to be a problem.

Reply:
it's used by AOTAutograd. If the compiler change the stride of any output tensors (e.g. saved activations) of the  fwd graph, we need make sure the bwd graph get the fx placeholder node with compiler generated stride (rather than eager stride) to be consistent. Code pointer here: 
 https://github.com/pytorch/pytorch/blob/308e154af5fd6388f49eabe631e7b78ca3ac9c39/torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py#L362

Reply:
Hi, this part of the code contains:
```python
if _get_symint_hints(ph_arg.stride()) != real_stride:
    placeholder_list[i] = ph_arg.as_strided(ph_arg.size(), real_stride)
```
I think ph_arg.stride() and real_stride have always been equivalent (mean if branch is always false), I notice this snippet write before #104971, thus, after compiling backward immediately, wouldn't there be no such inequality? If there are any examples of how the two would indeed be different, please let me know

Reply:
`ph_arg.stride()` represents the stride of placeholder if we are running eager model; `real_stride` represents the stride of the placeholder if we run the compiled model. The compiler may pick different strides for tensors for performance . So these 2 may not match.

One example is regarding convolution. We will convert tensors to channels last if that is likely to help. Maybe some tests here https://github.com/pytorch/pytorch/blob/main/test/inductor/test_layout_optim.py can demonstrate this.